src/main/io/gps.h

```diff
@@ -131,6 +165,12 @@ typedef struct gpsSolutionData_s {
     uint8_t numSat;
 } gpsSolutionData_t;
 
+typedef struct ubxMonVer_s {
+    char swVersion[30];
+    char hwVersion[10];
+    char extension[10 * 30];
+} ubxMonVer_t;
```

Comment:
340 bytes of RAM is considerable in BF. It will also make other structures larger.
IMO not worth it (at least ignore extension)

Reply:
is it possible to have this volatile and delete it after read?

Reply:
Current ublox packet is larger, so no problem there. But this info is store for CLI (and only for cli), it is too wasteful.

Maybe CLI can repeat ubxMonVer_t request and print it?

Rest of code needs only hwVersion and it can be converted to integer.

Reply:
Late to the party, but I wanted to think more about this.

I agree that this is substantial amount of RAM, which should be fine as long as we _need_ the data.

But we don't - the FC only actually needs two types of messages (DOP, PVT) which fit into 92 bytes.

If we really want to auto-detect the GPS version (rather than asking the user to select it from a dropdown in Configurator) then we can get away with the first few bytes of MON-VER.

The rest (e.g., NAV-SAT) are not needed on the FC at all: they are never understood / displayed by the FC, these are purely for the aesthetic display of the _Configurator_ GPS screen, so moving the parsing code of those messages to the Configurator would not give us most of this memory back (and would allow the configurator to fully process the message, rather than arbitrarily truncating at 32 satellites).

Overall, there is no need to allocate 300+ bytes to process the first few bytes of VER, and we should not process any messages on the FC that are not directly useful for the flight.

Likely not in this PR, but it should be fixed in a follow-up.

FYI @freasy @ledvinap